forte/data/data_pack.py

```diff
@@ -1399,11 +1399,14 @@ def _generate_link_entry_data(
                     f"{a_type}"
                 )
 
+            link_parent = cast(Entry, link.parent)
             a_dict["parent"].append(
-                np.where(data[parent_type]["tid"] == link.parent)[0][0]
+                np.where(data[parent_type]["tid"] == link_parent.tid)[0][0]
```

Comment:
Maybe we don't want to change the usage here. `link.parent` should be a `tid` instead of an entry object.